src/ObjectPool/test/DisposableObjectPoolTest.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Extensions.ObjectPool
+{
+    public class DisposableObjectPoolTest
+    {
+        [Fact]
+        public void DisposableObjectPoolWithDefaultPolicy_GetAnd_ReturnObject_SameInstance()
```

Comment:
Copied some tests from the `DefaultObjectPool` as the returning is handled by the `DisposableObjectPool`, so this should be tested.

Before the last commit it was just delegating to its base class, so no explicit tests were needed.